File: content/snippets/python/language_basics/stacks_and_queues.py
#!/usr/bin/env python3
from collections import deque


def main() -> None:
    # Stack: list used as LIFO
    stack: list[int] = []
    for v in [1, 2, 3]:
        stack.append(v)
    print("stack top:", stack[-1])  # 3
    stack.pop()
    print("after pop:", stack[-1])  # 2

    # Queue: deque used as FIFO
    queue: deque[int] = deque()
    for v in [1, 2, 3]:
        queue.append(v)
    print("queue front:", queue[0])  # 1
    queue.popleft()
    print("after dequeue:", queue[0])  # 2


if __name__ == "__main__":
    main()

File: content/snippets/go/language_basics/stacks_and_queues.go
package main

import "fmt"

func main() {
	// Stack: slice used as LIFO
	stack := []int{}
	for _, v := range []int{1, 2, 3} {
		stack = append(stack, v)
	}
	fmt.Println("stack top:", stack[len(stack)-1]) // 3
	stack = stack[:len(stack)-1]
	fmt.Println("after pop:", stack[len(stack)-1]) // 2

	// Queue: slice used as FIFO
	queue := []int{}
	for _, v := range []int{1, 2, 3} {
		queue = append(queue, v)
	}
	fmt.Println("queue front:", queue[0]) // 1
	queue = queue[1:]
	fmt.Println("after dequeue:", queue[0]) // 2
}
